src/commands/kv/namespace/site.rs

```diff
@@ -45,30 +44,16 @@ pub fn site(
                     failure::bail!("You will need to enable Workers Unlimited for your account before you can use this feature.")
                 } else if api_errors.errors.iter().any(|e| e.code == 10014) {
                     log::info!("Namespace {} already exists.", title);
-                    let params = ListNamespacesParams {
-                        page: Some(PAGE_NUMBER),
-                        per_page: Some(MAX_NAMESPACES_PER_PAGE),
-                    };
-
-                    let response = client.request(&ListNamespaces {
-                        account_identifier: &target.account_id,
-                        params: params,
-                    });
+                    let namespaces = kv::namespace::list::call_api(target, user)?;
```

Comment:
i really like this `::call_api` pattern; or even just `::call`. I'd like to incorporate this in other places too, i think it extends the work you did to divorce api calls from cli output. Should it be implemented for `namespace_create` as well to address the issues in the comment above? Seems like a similar refactor.

Reply:
I tried factoring out `::call_api` for `namespace::create`! :D The limitation I ran into was the fact that if site called `::call_api`, it would expect a `ApiError` instead of a `failure::Error`. Perhaps I can wrap the `ApiError` in a `failure:Error` and perform a match over it, so we can extract the `ApiError` and its error code for use by site.rs.

Reply:
hmm. do you think maybe that deserves its own separate analysis as part of client refactoring? it doesn't have to happen here.

Reply:
i think the pattern deserves some meta-level attention as far as its interface and what precisely it should take/return. we can defer if that makes more sense.

Reply:
I agree--this design decision is beginning to bleed into #63. We should think about what error type we specifically want ALL API calls to surface, and make those consistent. Shall we keep this PR on ice for now and use it to drive discussion?

Reply:
i'd say at least get in the `&GlobalUser` changes, I think those are solid. Your call on this part; if we're just going to churn we may as well hold off.

Reply:
I'll minimize this PR to just those `&GlobalUser` changes an open a new WIP one with a proposed approach to API error handling that we can discuss for #63 and #623.